node/forest_libp2p/src/service.rs

```diff
@@ -397,6 +268,411 @@ where
+                        network_sender_out,
+                        NetworkEvent::HelloRequest {
+                            request,
+                            source: peer,
+                        },
+                    )
+                    .await;
+                }
+                RequestResponseMessage::Response {
+                    request_id,
+                    response,
+                } => {
+                    // Send the sucessful response through channel out.
+                    if let Some(tx) = hello_request_table.remove(&request_id) {
+                        if tx.send(Ok(response)).is_err() {
+                            debug!("RPCResponse receive timed out");
+                        }
+                    } else {
+                        debug!("RPCResponse receive failed: channel not found");
+                    };
+                }
+            },
+            RequestResponseEvent::OutboundFailure {
+                peer,
+                request_id,
+                error,
+            } => {
+                debug!(
+                    "Hello outbound error (peer: {:?}) (id: {:?}): {:?}",
+                    peer, request_id, error
+                );
+
+                // Send error through channel out.
+                let tx = hello_request_table.remove(&request_id);
+                if let Some(tx) = tx {
+                    if tx.send(Err(error.into())).is_err() {
+                        debug!("RPCResponse receive failed");
+                    }
+                }
+            }
+            RequestResponseEvent::InboundFailure {
+                peer,
+                error,
+                request_id: _,
+            } => {
+                debug!("Hello inbound error (peer: {:?}): {:?}", peer, error);
+            }
+            RequestResponseEvent::ResponseSent { .. } => (),
+        },
+        ForestBehaviourEvent::Bitswap(bs_event) => {
+            let get_prefix = |query_id: &libp2p_bitswap::QueryId| {
+                if outgoing_bitswap_query_ids.contains(query_id) {
+                    "Outgoing"
+                } else {
+                    "Inbound"
+                }
+            };
+            match bs_event {
+                BitswapEvent::Progress(query_id, num_missing) => {
+                    let prefix = get_prefix(&query_id);
+                    trace!("{prefix} bitswap query {query_id} in progress, {num_missing} blocks pending");
+                }
+                BitswapEvent::Complete(query_id, result) => match result {
+                    Ok(()) => {
+                        let prefix = get_prefix(&query_id);
+                        outgoing_bitswap_query_ids.remove(&query_id);
+                        info!("{prefix} bitswap query {query_id} completed successfully");
+                        // TODO: Convert query_id to cid?
+                        // emit_event(&self.network_sender_out, NetworkEvent::BitswapBlock{query_id}).await;
```

Comment:
This should be the only thing that I cannot find a way to convert, to my understanding this is only for instrumentation. Any suggestions? @lemmih

Reply:
nvm this is fixed